File: Searching and Sorting/Binary Search/Binary_Search.py
n = 23
arr = [98,99,100,110,150,200,210,280]
low = 0
high = len(arr)-1
def binary_search(n,arr,low,high):
    mid = int((low+high)//2)
    if(n == arr[mid]):
        print('Found')
        return 0
    if(mid == low and mid == high and mid != n):
        print('Not Found')
    elif(n > arr[mid]):
        low = mid+1
        binary_search(n,arr,low,high)
    elif(n <arr[mid]):
        high = mid-1
        binary_search(n,arr,low,high)
    
binary_search(n,arr,low,high)

File: Searching and Sorting/Binary Search/BinarySearch.java

public class Main{
    public static int binarySearch(int[] a, int key){
        int low = 0;int high = a.length-1;
        while(low<=high){
            int mid = low+(high-low)/2;
            if (key < a[mid]) high = mid-1;
            else if(key > a[mid]) low = mid+1;
            else return mid;
        }
        return -1;
    }
    public static void main(String args[]){
        int a[] = {1,2,4,7,8,9,11,23,55,67,88};
        System.out.println(binarySearch(a,23));
    }
}
